torch/fx/graph_module.py

```diff
@@ -810,7 +809,8 @@ def __deepcopy__(self, memo):
         return res
 
     def __copy__(self):
-        res = GraphModule(self, self.graph)
+        from .lazy_graph_module import get_graph_module_cls
+        res = get_graph_module_cls()(self, self.graph)
```

Comment:
It's weird for a copy to return a different class
```suggestion
        res = self.__class__(self, self.graph)
```

Reply:
I'll make the change. The reason I do it in the first place is to ensure we get a LazyGraphModule by copying a (non lazy) GraphModule.  Now I've made sure PT2 stack generate LazyGraphModule in the first place, I'll make the change you suggested to make the behavior less confusing.

Reply:
hmm, it turns out we can not use self.__class__ to construct the copy.

self.__class__ actually is not GraphModule but GraphModuleImpl (defined in GraphModule.__new__). 

Using self.__class__ to construct the copy cause some test failures: 
```
python test/test_fx.py -k test_graph_module_replicate_for_dp
```

complaining we can not access _backend_hooks attribute from the return value of gm.__copy__() .  Looks like the returned GraphModule is not well constructed.

I'll revert this part of change just FYI, @peterbell10

Reply:
I guess that's why `__deepcopy__` does this strange dance then:
https://github.com/pytorch/pytorch/blob/dd5e9c64208bf4103abad4866c1f52bbdcd49c88/torch/fx/graph_module.py#L789

presumably that followed by an explicit `__init__` call would work.

Reply:
I didn't since I feel writing the code this way makes it a bit hard to understand. Also considering the original implementation of `GraphModule.__copy__` already return instance of a different class (since each GraphModule instance has its own class), I feel the current code may not look too bad. 

I can follow up if there is strong reason for the change.